### DISCUSSION TITLE
Can workflow_dispatch inputs be enhanced to set INPUT_* environmental variables?

### DISCUSSION THREAD

**Discussion by dogboydog** [community/community#25924] [Actions]
<p>HI,</p>
<p>I noticed that <code>INPUT_&lt;UPPERCASEDINPUTNAME&gt;</code> environmental variables are not set for inputs defined in a <code>workflow_dispatch</code> trigger.</p>
<p>The blog post says you define them the same way as action inputs, but it doesn’t explicitly say that the environmental variables will be set. Since it uses the action inputs documentation as a reference though, I almost expected them to work the same way.</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/" target="_blank" rel="nofollow noopener">The GitHub Blog</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/323;"><img src="https://user-images.githubusercontent.com/9920963/181094143-78c9e816-6b56-433e-ac12-e36ab72f67db.png" class="thumbnail" width="690" height="323" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b20e3cc5cd75d5f384f22ef8b13f292c738f5b11_2_690x323.png, https://aws1.discourse-cdn.com/github/original/2X/b/b20e3cc5cd75d5f384f22ef8b13f292c738f5b11.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/b20e3cc5cd75d5f384f22ef8b13f292c738f5b11.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b20e3cc5cd75d5f384f22ef8b13f292c738f5b11_2_10x10.png"></div>

<h3><a href="https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/" target="_blank" rel="nofollow noopener">GitHub Actions: Manual triggers with workflow_dispatch The GitHub Blog</a></h3>

<p>GitHub Actions: Manual triggers with workflow_dispatch</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>The documentation describes the format of the env variables for actions:<br>
<a href="https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs</a></p>
<p>Here is a run showing it is not set<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/dogboydog/manual-workflow-test/runs/912738471?check_suite_focus=true" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/9920963/181094144-dab4dfe7-d024-43d1-9c0a-8052b7d78f67.jpeg" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/dogboydog/manual-workflow-test/runs/912738471?check_suite_focus=true" target="_blank" rel="nofollow noopener">dogboydog/manual-workflow-test</a></h3>

<p>Testing new workflow_dispatch event in GitHub Actions - dogboydog/manual-workflow-test</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<br>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/dogboydog/manual-workflow-test/actions/runs/183599014/workflow" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/9920963/181094146-60307c81-73a6-4d85-97f1-6181e24b30e1.jpeg" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/dogboydog/manual-workflow-test/actions/runs/183599014/workflow" target="_blank" rel="nofollow noopener">dogboydog/manual-workflow-test</a></h3>

<p>Testing new workflow_dispatch event in GitHub Actions - dogboydog/manual-workflow-test</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Either way it can be set manually through the <code>echo"::set-env .... "</code> syntax so it’s not a huge deal. I just wondered if this was something that was considered.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/dogboydog">@dogboydog</a>,</p>
<p>Yes, currently GitHub does not automatically create environment variables for the inputs of the ‘<strong>workflow_dispatch</strong>’ trigger.</p>
<p>If you want to use the inputs as environment variables in the workflow, you can set the inputs as the workflow global environment variables via the ‘<a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#env" rel="nofollow noopener"><strong>env</strong></a>’ key.<br>
For example:</p>
<pre><code class="lang-yaml">on:
  workflow_dispatch:
    inputs:
      log_Level:
        description: 'Log level'
        required: true
        default: 'warning'

env:
  INPUT_LOG_LEVEL: ${{ github.event.inputs.log_Level }}

jobs:
  job1:
    runs-on: ubuntu-latest
    steps:
      - name: Print input
        run: echo "The log level is $INPUT_LOG_LEVEL"
</code></pre>
<p>However, as mentioned in the docs, you also need to note that:</p>
<blockquote>
<p>When more than one environment variable is defined with the same name, GitHub uses the most specific environment variable. For example, an environment variable defined in a step will override job and workflow variables with the same name, while the step executes. A variable defined for a job will override a workflow variable with the same name, while the job executes.</p>
</blockquote>
<p>If your projects really need the feature that GitHub can automatically create environment variables for the inputs of the ‘<strong>workflow_dispatch</strong>’ trigger, I recommend that you can directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>.<br>
That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by dogboydog** [comment]
<p>Thanks, I submitted the request.</p>

**Reply by Yarden-zamir** [reply]
Link it here so that people who get here can join

**Reply by dogboydog** [reply]
This was two years ago, I don't know how to find it.

**Reply by SvenStaehs** [reply]
well, I still stumbled over this three years later, so apparently it was not implemented 😆 (not on github.com though, on latest GHES)
